tools/lib/provision_inner.py

```diff
@@ -79,6 +80,16 @@ def configure_rabbitmq_paths() -> List[str]:
     return paths
 
 
+def check_environment() -> bool:
+    user_id = os.getuid()
+    user_name = pwd.getpwuid(user_id).pw_name
+    wsl_path = "/proc/sys/fs/binfmt_misc/WSLInterop"
```

Comment:
Since most WSL distributions has `Interoperability` enabled, I think it's an appropriate path to check for, we can also use windows mounted disc paths such as `/mnt/c/Users` or `/mnt/d`  to identify the machine as WSL. I'm not sure about which one would be more appropriate in the context of this problem.